admin/class-acc_user_importer-admin.php

```diff
@@ -510,8 +525,93 @@ private function proccess_user_data ( $users ) {
 		
 		return $api_response;
 	}
+
+
+	/**
+	 * Returns True if the user is expired.
+	 * If the user has no 'expiry' field, it must be a stale entry in the database.
+	 * So it is considered expired.
+	 */
+	private function is_user_expired ($user) {
+		if (empty($user->expiry)) {
+			$this->log_dual("user $user->ID $user->display_name has no expiry, consider expired");
+			return true;
+		}
+
+		if ($user->expiry < date("Y-m-d")) {
+			//$this->log_dual("user $user->ID $user->display_name expiry=$user->expiry is expired");
+			return true;
+		}
+		//$this->log_dual("user $user->ID $user->display_name expiry=$user->expiry is valid");
+		return false;
+	}
 	
 
+	/**
+	 * Go over our local user database and see who has an expired membership.
+	 */
+	private function proccess_expiry () {
+		$GLOBALS['acc_logstr'] = "";		//Clear the API response log string
+
+		// Get user-configurable option values
+		$options = get_option('accUM_data');
+
+		$this->log_dual("Now looking for expired members.");
+
+		//create response object
+		$api_response = Array();
```

Comment:
Not strictly required, but the `array()` syntax is considered outdated and I'd generally recommend using `[]` to create a new array.

Reply:
Sure, I will change it.